unified_incoming_viewing_keys.py

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env python3
+import sys; assert sys.version_info[0] >= 3, "Python 3 required."
+
+from random import Random
+
+from cryptography.hazmat.primitives.asymmetric import ec
+from cryptography.hazmat.primitives.serialization import PublicFormat, Encoding
+
+from zcash_test_vectors.output import render_args, render_tv, Some
+from zcash_test_vectors.rand import Rand, randbytes
+from zcash_test_vectors.orchard import key_components as orchard_key_components
+from zcash_test_vectors.sapling import zip32 as sapling_zip32
+from zcash_test_vectors.unified_encoding import encode_unified, decode_unified
+from zcash_test_vectors.unified_encoding import P2PKH_ITEM, SAPLING_ITEM, ORCHARD_ITEM
+
+def main():
+    args = render_args()
+
+    rng = Random(0xabad533d)
+    rand = Rand(randbytes(rng))
+    seed = rand.b(32)
+
+    test_vectors = []
+    for i in range(0, 10):
+        has_t_key = rand.bool()
+        if has_t_key:
+            c = rand.b(32)
+            privkey = ec.derive_private_key(int.from_bytes(rand.b(32), 'little'), ec.SECP256K1())
+            pubkey = privkey.public_key()
+            pubkey_bytes = pubkey.public_bytes(Encoding.X962, PublicFormat.CompressedPoint)
+            assert len(pubkey_bytes) == 33
+            assert pubkey_bytes[0] in (0x02, 0x03)
+            t_key_bytes = c + pubkey_bytes
+        else:
+            t_key_bytes = None
+
+        has_s_key = rand.bool()
+        if has_s_key:
+            root_key = sapling_zip32.ExtendedSpendingKey.master(seed)
+            purpose_key = root_key.child(sapling_zip32.hardened(32))
+            coin_key = purpose_key.child(sapling_zip32.hardened(133))
+            account_key = coin_key.child(sapling_zip32.hardened(i))
```

Comment:
Post-hoc: why does the Sapling key use correct ZIP-316 derivation, while the Orchard and transparent components just use random keys?

Reply:
I think it's because there is no implementation in this repo of master key generation for Orchard or transparent. The same comment applies to `unified_full_viewing_keys.py`. I'll consider fixing this when I implement the Orchard ZIP 32 test vectors.